### PR TITLE
nrf91: Update GPS to GNSS

### PR DESCRIPTION
**Pull request by JianyuWang0623**
## Summary
Update GPS to GNSS.
- Config
    - SENSORS_GPS => SENSORS_GNSS
- Header
    - nuttx/sensors/gps.h => nuttx/sensors/gnss.h
- Struct
    - gps_lowerhalf_s => gnss_lowerhalf_s
    - gps_ops_s => gnss_ops_s
    - sensor_gps => sensor_gnss
    - sensor_gps_satellite => sensor_gnss_satellite
- Macro
    - SENSOR_GPS_SAT_INFO_MAX => SENSOR_GNSS_SAT_INFO_MAX
    - SENSOR_TYPE_GPS_SATELLITE => SENSOR_TYPE_GNSS_SATELLITE
    - SENSOR_TYPE_GPS => SENSOR_TYPE_GNSS
- Function
    - gps_register() => gnss_register()
- Related: 03f4ec7765ed8b3551f1a2bc8198c5007a542788 (A mistake that not using the lasted version while rename GPS to GNSS, missing 42211cc1c3013431dbc03c4db49545b97a0574cd maybe)

## Impact
nrf91: arch/boards

## Testing
```
./tools/configure.sh -l nrf9160-dk:modem_ns
```

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13494&type=pullrequests)



The PR appears to meet the basic NuttX requirements, but could benefit from more details in a few areas:

**Strengths:**

* **Summary:** Clearly outlines the "Why," "What," and "How" of the change.  Good use of bullet points for readability.
* **Impact:**  Identifies the specific area of impact (nrf91: arch/boards).
* **Testing:**  Provides a specific configuration command used for testing.

**Areas for Improvement:**

* **Impact:**
    * **Feature Changes:**  Be more explicit. Is GNSS a *replacement* for GPS, or an *addition*?  Will existing GPS functionality break?
    * **User Impact:** Does this change how users interact with the GPS/GNSS API? Any new configuration options?
    * **Build Impact:**  Does the `configure.sh` command need any modifications due to this change?
    * **Hardware Impact:**  Which specific boards are affected within `arch/boards/nrf91`? 
    * **Documentation Impact:**  Will documentation need updates to reflect the GNSS changes? If so, is this update included in the PR?
    * **Other Impacts:** Briefly address the remaining impact categories.  Even a simple "NO" is better than leaving them blank. 
* **Testing:**
    * **Build/Target Details:** Provide more information about your testing environment:
        * Build Host OS, CPU Architecture, Compiler 
        * Target Architecture (simulator or real hardware?), Specific board and configuration used 
    * **Testing Logs:** The current logs are placeholders.  Include *actual* log snippets from *before* and *after* the change. Focus on demonstrating that the GNSS functionality works as intended.

**Recommendation:**

Expand on the "Impact" and "Testing" sections with the details mentioned above. This will make your PR stronger and easier for reviewers to assess.

File: libs/libc/gnssutils/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /minmea
+db46128e73cee26d6a6eb0482dcba544ee1ea9f5.zip
```
Comment (xiaoxiang781216) [inline libs/libc/gnssutils/.gitignore:2]: *.zip

Comment (JianyuWang0623) [reply libs/libc/gnssutils/.gitignore]: Done

**Comment by JianyuWang0623** [comment]
Details for 622ddb53da9fc122a832985cabaface7c1a9ee7c:
libs/libc/gnssutils/Make.defs will delete *.zip:
```
	$(call DELFILE, $(MINMEA_VERSION).zip)
```
but libs/libc/gnssutils/CMakeLists.txt may not

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**